solr/core/src/java/org/apache/solr/util/CryptoKeys.java

```diff
@@ -295,15 +304,28 @@ public PublicKey getPublicKey() {
       return publicKey;
     }
 
-    public byte[] encrypt(ByteBuffer buffer) {
```

Comment:
We only call `ByteBuffer.wrap` in all invocations of this so isn't helpful to have this as a `ByteBuffer`. The decryption also just uses a plain `byte[]` so this makes it consistent.

Reply:
This is a public method on a public utility class though; I wonder if the intention was to have this used flexibly from elsewhere? This could in principle break plugin code ...

This change seems good still, but worth explicitly calling out this possible concern.

Reply:
Yea I'm just going to back out this change.

Reply:
Reverted this in 741e131ebbc43b7e49fc668e0c76cb6163946b3b